Tree Split using bitwise instead of boolean operands

#### Issue description

From the CRAN package check page, there is a WARN being issued: 

```
Found the following significant warnings:
  ./mlpack/core/tree/rectangle_tree/x_tree_split_impl.hpp:152:17: warning: use of bitwise '&' with boolean operands [-Wbitwise-instead-of-logical]
  ./mlpack/core/tree/rectangle_tree/x_tree_split_impl.hpp:167:19: warning: use of bitwise '&' with boolean operands [-Wbitwise-instead-of-logical]
```

https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-debian-clang/mlpack-00check.html

https://github.com/mlpack/mlpack/blob/9f0bd1aa234229835f070a9dd201c3871c88fb49/src/mlpack/core/tree/rectangle_tree/x_tree_split_impl.hpp#L152-L153

https://github.com/mlpack/mlpack/blob/9f0bd1aa234229835f070a9dd201c3871c88fb49/src/mlpack/core/tree/rectangle_tree/x_tree_split_impl.hpp#L167-L168

#### Your environment

 * version of mlpack: v3.4.1
 * operating system: Debian GNU/Linux testing
 * compiler: clang version 14.0.6-2; GNU Fortran (GCC) 11.3.0 (Debian 11.3.0-5)
 * version of dependencies (Boost/Armadillo): 
 * any other environment information you think is relevant: R bindings

--- a/src/mlpack/core/tree/rectangle_tree/x_tree_split_impl.hpp
+++ b/src/mlpack/core/tree/rectangle_tree/x_tree_split_impl.hpp
@@ -149,7 +149,7 @@ bool XTreeSplit::SplitNonLeafNode(TreeType *tree, std::vector<bool>& relevels)
   for (size_t i = lastDim + 1; i < axes.size(); ++i)
   {
     for (size_t j = 0; j < tree->NumChildren(); ++j)
-      axes[i] = axes[i] &
+      axes[i] = axes[i] &&
           tree->Child(j).AuxiliaryInfo().SplitHistory().history[i];
     if (axes[i] == true)
     {
@@ -164,7 +164,7 @@ bool XTreeSplit::SplitNonLeafNode(TreeType *tree, std::vector<bool>& relevels)
     {
       axes[i] = true;
       for (size_t j = 0; j < tree->NumChildren(); ++j)
-        axes[i] = axes[i] &
+        axes[i] = axes[i] &&
                   tree->Child(j).AuxiliaryInfo().SplitHistory().history[i];
       if (axes[i] == true)
       {

